server/server.go

```diff
@@ -176,59 +215,177 @@ func (s *server) Boot() {
 			// Combine all options this server defines.
 			options := []kithttp.ServerOption{
 				kithttp.ServerBefore(s.RequestFuncs()...),
-				kithttp.ServerErrorEncoder(s.ErrorEncoder()),
+				kithttp.ServerErrorEncoder(errorEncoder),
 			}
 
 			// Register all endpoints to the router depending on their HTTP methods and
 			// request paths. The registered http.Handler is instrumented using
 			// prometheus. We track counts of execution and duration it took to complete
 			// the http.Handler.
 			s.router.Methods(method).Path(path).Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				var err error
```

Comment:
Slightly concerned about the length of this method, as well as it including both transactions and metrics. No blocker though, but we might want to try and refactor this in the future.